app/streaming/video/ffmpeg-renderers/eglvid.cpp

```diff
@@ -0,0 +1,480 @@
+// vim: noai:ts=4:sw=4:softtabstop=4:expandtab
+#define GL_GLEXT_PROTOTYPES
+
+#include "eglvid.h"
+
+#include "streaming/session.h"
+#include "path.h"
+
+#include <QDir>
+
+#include <Limelight.h>
+#include <unistd.h>
+
+#include <SDL_egl.h>
+#include <SDL_opengl.h>
+#include <SDL_syswm.h>
+#include <SDL_opengles2.h>
+#include <SDL_opengles2_gl2ext.h>
+
+/* TODO:
+ *  - handle more pixel formats
+ *  - code refacto/cleanup
+ *  - handle software decoding
+ *  - handle window resize
+ */
+
+/* DOC/misc:
+ *  - https://kernel-recipes.org/en/2016/talks/video-and-colorspaces/
+ *  - http://www.brucelindbloom.com/
+ *  - https://learnopengl.com/Getting-started/Shaders
+ *  - https://github.com/stunpix/yuvit
+ *  - https://en.wikipedia.org/wiki/YCbCr#ITU-R_BT.601_conversion
+ *  - https://www.renesas.com/eu/en/www/doc/application-note/an9717.pdf
+ *  - https://www.xilinx.com/support/documentation/application_notes/xapp283.pdf
+ *  - https://www.itu.int/dms_pubrec/itu-r/rec/bt/R-REC-BT.709-6-201506-I!!PDF-E.pdf
+ *  - https://www.khronos.org/registry/OpenGL/extensions/OES/OES_EGL_image_external.txt
+ *  - https://gist.github.com/rexguo/6696123
+ *  - https://wiki.libsdl.org/CategoryVideo
+ */
+
+static QStringList egl_get_extensions(EGLDisplay dpy) {
+    const auto egl_extensions_str = eglQueryString(dpy, EGL_EXTENSIONS);
+    if (!egl_extensions_str) {
+        SDL_LogWarn(SDL_LOG_CATEGORY_APPLICATION, "Unable to get EGL extensions");
+        return QStringList();
+    }
+    return QString(egl_extensions_str).split(" ");
+}
+
+EGLExtensions::EGLExtensions(EGLDisplay dpy) :
+    m_extensions(egl_get_extensions(dpy))
+{}
+
+bool EGLExtensions::is_supported(const QString &extension) const {
```

Comment:
Can you move this `EGLExtensions` stuff into a separate file? It doesn't feel like it belongs here, since it's declared in `renderer.h`.